configure.ac

```diff
@@ -231,6 +231,12 @@ LIBMESH_TEST_CXX11_NOEXCEPT
 #-----------------------------------------------------
 LT_INIT([disable-static])
 
+# We need to set the no-undefined flag for linking 
+# with libtool, because Windows requires all shared
+# libraries to include all used symbols. This flag
+# should do no harm on other platforms.
+libmesh_LDFLAGS+="-no-undefined"
```

Comment:
The flag -no-undefined is a libtool flag. It looks like it is passed directly to the liker without libtool.